effects/populate-cache/default.nix

```diff
@@ -0,0 +1,179 @@
+{inputs, withSystem, ...}:
+let
+  attic-client = inputs.attic.packages."x86_64-linux".attic-client;
+  cachix = withSystem "x86_64-linux" ({pkgs, ...}: pkgs.cachix);
+  in
+{
+  inputs,
+  lib,
+  withSystem,
+  config,
+  ...
+}: {
+  imports = [
+    inputs.hercules-ci-effects.flakeModule
+  ];
+
+  options = {
+    populate-cache-effect = {
+      enable = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = "Enables HerculesCI effects populating some external cache.";
+      };
+      attic-client-pkg = lib.mkOption {
+        type = lib.types.package;
+        description = "Version of the attic-client package to use on \"x86_64-linux\".";
+        default = attic-client;
+      };
+      cachix-pkg = lib.mkOption {
+        type = lib.types.package;
+        description = "Version of the cachix package to use on \"x86_64-linux\".";
+        default = cachix;
+      };
+      caches = lib.mkOption {
+        description = "
+          An attribute set, each `name: value` pair translates to an effect under
+          onPush.default.outputs.effects.populate-cache-effect.name
+        ";
+        example = "
+          {
+            our-cachix = {
+              type = \"cachix\";
+              secretName = \"our-cachix-token\";
+              branches = [ \"master\" ];
+              packages = [ pkgs.hello ];
+            };
+          }
+        ";
+        type = lib.types.attrsOf (lib.types.submodule (
+          {name, ...}: {
+            options = {
+              name = lib.mkOption {
+                type = lib.types.str;
+                default = name;
+                description = ''
+                  Name of the effect. By default it's the attribute name.
+                '';
+              };
+              type = lib.mkOption {
+                type = lib.types.enum ["attic" "cachix"];
+                description = "A string \"attic\" or \"cachix\".";
+              };
+              packages = lib.mkOption {
+                type = with lib.types; listOf package;
+                description = "List of packages to push to the cache.";
+                example = "[ pkgs.hello ]";
+              };
+              secretName = lib.mkOption {
+                type = lib.types.str;
+                description = ''
+                  Name of the HerculesCI secret. See [HerculesCI docs](https://docs.hercules-ci.com/hercules-ci-agent/secrets-json).
+                  The secrets "data" field should contain given data:
+
+                  ```
+                    "data": {
+                      "name": "my-cache-name",
+                      "token": "ey536428341723812",
+                      "endpoint": "https://my-cache-name.com"
+                    }
+                  ```
+
+                  The "endpoint" field is needed for Attic cache. With Cachix cache the "endpoint" field is not read and can be absent.
+                '';
+              };
```

Comment:
I hadn't even asked myself this search problem, I defer to your judgement :)

Reply:
I would leave without a default as otherwise there is still strong coupling to an outside piece of configuration only it's hidden away, which is worse. I find myself spending a lot of time figuring out these hidden globals in various systems.

Reply:
Resolved: leave it as is.